docs/bugs.json

```diff
@@ -1,21 +1,22 @@
 [
-	{
-		"name": "EventStructWrongData",
-		"summary": "Using structs in events logged wrong data.",
-		"description": "If a struct is used in an event, the address of the struct is logged instead of the actual data.",
-		"introduced": "0.4.17",
-		"fixed": "0.5.0",
-		"severity": "very low"
-	},
-	{
-		"name": "NestedArrayFunctionCallDecoder",
-		"summary": "Calling functions that return multi-dimensional fixed-size arrays can result in memory corruption.",
-		"description": "If Solidity code calls a function that returns a multi-dimensional fixed-size array, array elements are incorrectly interpreted as memory pointers and thus can cause memory corruption if the return values are accessed. Calling functions with multi-dimensional fixed-size arrays is unaffected as is returning fixed-size arrays from function calls. The regular expression only checks if such functions are present, not if they are called, which is required for the contract to be affected.",
-		"introduced": "0.1.4",
-		"fixed": "0.4.22",
-		"severity": "medium",
-		"check": {"regex-source": "returns[^;{]*\\[\\s*[^\\] \\t\\r\\n\\v\\f][^\\]]*\\]\\s*\\[\\s*[^\\] \\t\\r\\n\\v\\f][^\\]]*\\][^{;]*[;{]"}
-	},
+    {
+        "name": "EventStructWrongData",
+        "summary": "Using structs in events logged wrong data.",
+        "description": "If a struct is used in an event, the address of the struct is logged instead of the actual data.",
+        "introduced": "0.4.17",
+        "fixed": "0.5.0",
+        "severity": "very low",
+        "check": {"json-path": {"query": "$..[?(@.name === 'EventDefinition')]..[?(@.name === 'UserDefinedTypeName' && @.attributes.type.startsWith('struct'))]", "astType": "ast-json"}}
```

Comment:
Now that I see this - wouldn't it be easier to use the compact json ast?

Reply:
Updated it. It's kinda the same in the end (unless I missed a way to make it shorter)